src/test_utils/hive/mod.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::HashMap;
+
+use ef_testing::evm_sequencer::account::{AccountType, KakarotAccount};
+use ethers::types::U256 as EthersU256;
+use eyre::eyre;
+use katana_primitives::{
+    contract::ContractAddress,
+    genesis::json::{GenesisContractJson, GenesisJson},
+};
+use reth_primitives::{Address, Bytes, B256, U256, U64};
+use serde::{Deserialize, Serialize};
+use starknet::core::utils::get_storage_var_address;
+use starknet_api::core::ClassHash;
+use starknet_crypto::FieldElement;
+
+use super::katana::genesis::{KatanaGenesisBuilder, Loaded};
+
+/// Types from https://github.com/ethereum/go-ethereum/blob/master/core/genesis.go#L49C1-L58
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct HiveGenesisConfig {
+    pub config: Config,
+    pub coinbase: Address,
+    pub difficulty: U64,
+    pub extra_data: Bytes,
+    pub gas_limit: U64,
+    pub nonce: U64,
+    pub timestamp: U64,
+    pub alloc: HashMap<Address, AccountInfo>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Config {
+    pub chain_id: i128,
+    pub homestead_block: i128,
+    pub eip150_block: i128,
+    pub eip150_hash: B256,
+    pub eip155_block: i128,
+    pub eip158_block: i128,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct AccountInfo {
+    pub balance: U256,
+    pub code: Option<Bytes>,
+    pub storage: Option<HashMap<U256, U256>>,
+}
+
+impl HiveGenesisConfig {
+    /// Convert the [HiveGenesisConfig] into a [GenesisJson] using an [KatanaGenesisBuilder]<[Loaded]>. The [Loaded]
+    /// marker type indicates that the Kakarot contract classes need to have been loaded into the builder.
+    pub fn try_into_genesis_json(self, builder: KatanaGenesisBuilder<Loaded>) -> Result<GenesisJson, eyre::Error> {
+        let coinbase_address = FieldElement::from_byte_slice_be(self.coinbase.as_slice())?;
+        let builder = builder.with_kakarot(coinbase_address)?;
+
+        // Get the current state of the builder.
+        let kakarot_address = builder.cache_load("kakarot_address")?;
+        let proxy_class_hash = ClassHash(builder.proxy_class_hash()?.into());
```

Comment:
why can those getters return errors? since in the Loaded status, we are sure we have loaded the class hashes?

Reply:
If there was an error at loading. The getter returns the value from the mapping and converts the option into an error if the value is missing.